### PR TITLE
[devices.py] Refactor method of determining whether a service is fully started

### PR DESCRIPTION
**Pull request by jleveque**
### Summary

Instead of checking for the whether a `docker wait <service_name>` command is running to determine whether a service is fully started, instead we check whether or not the container itself is running using `docker inspect -f '{{.State.Running}}' <service_name>`.

Fixes https://github.com/Azure/sonic-mgmt/issues/1215

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### How did you do it?

Instead of checking for the whether a `docker wait <service_name>` command is running to determine whether a service is fully started, instead we check whether or not the container itself is running using `docker inspect -f '{{.State.Running}}' <service_name>`.

### CODE REVIEW

**Comment by jleveque** [comment]
@yvolynets-mlnx: Can you please confirm that this PR addresses https://github.com/Azure/sonic-mgmt/issues/1215?

File: tests/common/devices.py [hunk 1]
```diff
@@ -136,15 +136,15 @@ def is_service_fully_started(self, service):
...
+        This function assumes that the final step of all services checked by this function is to spawn a Docker
+        container with the same name as the service. We determine that the service has fully started if the
+        Docker container is running.
 
         @param service: Name of the SONiC service
         """
         try:
-            output = self.command('pgrep -f "docker wait %s"' % service)
-            if output["stdout_lines"]:
+            output = self.command("docker inspect -f '{{.State.Running}}' %s" % service)
```
Comment (wangxin) [inline tests/common/devices.py]: Tested this fix, ansible complains:
```
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:79 [arc-switch1029] command: {'_raw_params': "docker inspect -f '{{.State.Running}}' swss"}
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:143 {"unreachable": {}, "contacted": {"arc-switch1029": {"msg": "ERROR! template error while templating string: unexpected '.'", "failed": true}}}
ERROR    root:devices.py:151 Failed to get service status, exception: RunAnsibleModuleFail('run module command failed, errmsg {\'msg\': u"ERROR! template error while templating string: unexpected \'.\'", \'failed\': True}',)
```

It works after replaced this line to below:
```
output = self.command("docker inspect -f \{\{.State.Running\}\} %s" % service)

```

Log:
```
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:79 [arc-switch1029] command: {'_raw_params': 'docker inspect -f \\{\\{.State.Running\\}\\} swss'}
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:143 {"unreachable": {}, "contacted": {"arc-switch1029": {"cmd": ["docker", "inspect", "-f", "{{.State.Running}}", "swss"], "end": "2019-11-19 03:30:19.086603", "_ansible_no_log": false, "stdout": "true", "changed": true, "start": "2019-11-19 03:30:18.930894", "delta": "0:00:00.155709", "stderr": "", "rc": 0, "invocation": {"module_name": "command", "module_args": {"creates": null, "executable": null, "chdir": null, "_raw_params": "docker inspect -f \\{\\{.State.Running\\}\\} swss", "removes": null, "warn": true, "_uses_shell": false}}, "stdout_lines": ["true"], "warnings": []}}}
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:79 [arc-switch1029] command: {'_raw_params': 'docker inspect -f \\{\\{.State.Running\\}\\} syncd'}
DEBUG    pytest_ansible.module_dispatcher.v2:v2.py:143 {"unreachable": {}, "contacted": {"arc-switch1029": {"cmd": ["docker", "inspect", "-f", "{{.State.Running}}", "syncd"], "end": "2019-11-19 03:30:20.608447", "_ansible_no_log": false, "stdout": "false", "changed": true, "start": "2019-11-19 03:30:20.500474", "delta": "0:00:00.107973", "stderr": "", "rc": 0, "invocation": {"module_name": "command", "module_args": {"creates": null, "executable": null, "chdir": null, "_raw_params": "docker inspect -f \\{\\{.State.Running\\}\\} syncd", "removes": null, "warn": true, "_uses_shell": false}}, "stdout_lines": ["false"], "warnings": []}}}
```

**Comment by yvolynets-mlnx** [comment]
> 
> 
> @yvolynets-mlnx: Can you please confirm that this PR addresses #1215?

Yes, it is.

File: tests/common/devices.py (hunk 1 above)
Comment (jleveque) [reply tests/common/devices.py]: Thanks for the info, Xin! I have pushed another commit to escape the curly braces.

**Pull request merged**